src/modules/navigator/navigator_main.cpp

```diff
@@ -640,6 +641,11 @@ Navigator::task_main()
 		case vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION:
 			_pos_sp_triplet_published_invalid_once = false;
 			navigation_mode_new = &_mission;
+
+			if (_navigation_mode == &_mission_reverse) {
+				_mission.switch_from_reverse();
```

Comment:
This should probably be contained within MisisonReverse inactivation(). It's easier to reason if the regular Mission can be kept in tact at all times except the special case when MissionReverse is actually running.

Reply:
Actually the problem here is the order of execution. As the on_activation() from Mission is executed before on_inactivation of MissionReverse I had to add this flag.